File: walletfreak_app/src/api/endpoints/cards.ts
import apiClient from '../client';

export interface CardListParams {
  page?: number;
  issuer?: string;
  category?: string;
  min_fee?: number;
  max_fee?: number;
  search?: string;
  sort?: string;
  wallet?: string;
}

export const getCards = async (params?: CardListParams) => {
  const { data } = await apiClient.get('/cards/', { params });
  return data;
};

export const getCardDetail = async (slug: string) => {
  const { data } = await apiClient.get(`/cards/${slug}/`);
  return data;
};

File: walletfreak/api/schemas/cards.py
from ninja import Schema
from typing import Optional


class CardListParams(Schema):
    page: int = 1
    page_size: int = 20
    issuer: Optional[str] = None
    category: Optional[str] = None
    min_fee: Optional[int] = None
    max_fee: Optional[int] = None
    search: Optional[str] = None
    sort: Optional[str] = None
    wallet: Optional[str] = None
